tests/test_simple_target_loop.py

```diff
@@ -0,0 +1,188 @@
+                             "Scheduler in %s, expected OFFLINE. " % salobj.State(self.scheduler.summary_state))
+
+            # now enable queue...
+            await self.enable_queue()
+
+            # ...and try again. This time the scheduler should stay in enable and publish targets to the queue.
+
+            def assertEnable(data):
+                """Callback function to make sure scheduler is enabled"""
+                self.assertEqual(data.summaryState, salobj.State.ENABLED,
+                                 "Scheduler unexpectedly transitioned from "
+                                 "ENABLE to %s" % salobj.State(data.summaryState))
+
+            # I will wait for the scheduler to publish 2 targets and then I will disable it
+            def count_targets(data):
+                """Callback to count received targets"""
+                self.received_targets += 1
+
+            def count_heartbeats(data):
+                """Callback to count heartbeats"""
+                self.heartbeats += 1
+
+            # Subscribing callbacks for the test
+            self.scheduler_remote.evt_target.callback = count_targets
+            self.scheduler_remote.evt_heartbeat.callback = count_heartbeats
+
+            await self.enable_scheduler()
+
+            self.scheduler_remote.evt_summaryState.callback = assertEnable
+
+            # Need to time this test and timeout if it takes took long
+            start_time = time.time()
+            while self.received_targets < self.expected_targets:
+                if time.time() - start_time > self.target_test_timeout:
+                    break
+                await asyncio.sleep(10)
+
+            # Test over, unsubscribe callbacks
+            self.scheduler_remote.evt_summaryState.callback = None
+            self.scheduler_remote.evt_target.callback = None
+            self.scheduler_remote.evt_heartbeat.callback = None
+            end_time = time.time()
+
+            # Test completed, disabling scheduler
+            cmd_attr = getattr(self.scheduler_remote, f"cmd_disable")
+            await cmd_attr.start(cmd_attr.DataType())
+
+            # Scheduler should be in OFFLINE state.
+            self.assertEqual(self.scheduler.summary_state, salobj.State.DISABLED,
+                             'Scheduler in %s, expected %s' % (self.scheduler.summary_state,
+                                                               salobj.State.DISABLED))
+
+            # Must have received at least the expected number of targets
+            self.assertGreaterEqual(self.received_targets, self.expected_targets,
+                                    'Failed target production loop. Got %i of %i' % (self.received_targets,
+                                                                                     self.expected_targets))
+
+            expected_heartbeats = int((end_time - start_time)/salobj.base_csc.HEARTBEAT_INTERVAL)
```

Comment:
Perhaps I should make the heartbeat interval an attribute of BaseCsc? (I could also make HEARTBEAT_INTERVAL a public symbol, but that seems a bit less safe since a subclass might want to override the value).

Reply:
I do like the idea of HEARTBEAT_INTERVAL as an attribute of BaseCsc.